src/lib/registration/registration_ui.rb

```diff
@@ -181,8 +218,174 @@ def disable_update_repos(product_service)
+              "in order to receive the latest updates?")
+        else # Yast::Mode.update
+          # TRANSLATORS: updates popup question (2/2), multiline, max. ~60 chars/line
+          msg += _("Would you like to enable these repositories during upgrade\n" \
+              "in order to receive the latest updates?")
+        end
+
+        options.install_updates = Yast::Popup.YesNo(msg)
+      end
+
+      options.install_updates
+    end
+
     private
 
     attr_accessor :registration
+
+    def register_system
+      options = Storage::InstallationOptions.instance
+      base_product = SwMgmt.find_base_product
+      distro_target = base_product["register_target"]
+
+      log.info "Registering system, distro_target: #{distro_target}"
+
+      Yast::Popup.Feedback(_(CONTACTING_MESSAGE),
+        _("Registering the System...")) do
+        registration.register(options.email, options.reg_code, distro_target)
+      end
+    end
+
+    # the credentials are read from Storage::InstallationOptions
+    def register_base_product
+      options = Storage::InstallationOptions.instance
+      return if options.base_registered
+
+      # then register the product(s)
+      base_product = SwMgmt.find_base_product
+
+      Yast::Popup.Feedback(_(CONTACTING_MESSAGE),
+        _("Registering %s ...") % SwMgmt.product_label(base_product)
+      ) do
+        base_product_data = SwMgmt.base_product_to_register
+        base_product_data["reg_code"] = options.reg_code
+        registration.register_product(base_product_data, options.email)
+      end
+    end
+
+    # register all selected addons
+    def register_selected_addons(selected_addons, known_reg_codes)
+      # create duplicate as array is modified in loop for registration order
+      registration_order = selected_addons.clone
+
+      product_succeed = registration_order.map do |product|
+        ConnectHelpers.catch_registration_errors(
+          message_prefix: "#{product.label}\n") do
+          register_selected_addon(product, known_reg_codes[product.identifier])
+        end
+
+        # remove from selected after successful registration
+        selected_addons.reject! { |selected| selected.identifier == product.identifier }
+      end
+
+      !product_succeed.include?(false) # succeed only if noone failed
+    end
+
+    def register_selected_addon(product, reg_code)
+      product_service = Yast::Popup.Feedback(
+        _(CONTACTING_MESSAGE),
+        # %s is name of given product
+        _("Registering %s ...") % product.label) do
+        product_data = {
+          "name"     => product.identifier,
+          "reg_code" => reg_code,
+          "arch"     => product.arch,
+          "version"  => product.version
+        }
+
+        registration.register_product(product_data)
+      end
+
+      # select repositories to use in installation (e.g. enable/disable Updates)
+      select_repositories(product_service) if Yast::Mode.installation || Yast::Mode.update
```

Comment:
I have checked this. Yast::Mode.installation also includes AutoYaST case. So it is no problem....